lib/tools.js

```diff
@@ -2223,14 +2223,14 @@ function maybeCallback(callback, ...args) {
 /**
  * Checks if the given callback is a function and if so calls it with the given error and parameter immediately, else a resolved or rejected Promise is returned
  *
- * @param {function(Error=, ...any=):void|undefined} callback - callback function to be executed
- * @param {any} [error] - error which will be used by the callback function. If callback is not a function and
+ * @param {((error: Error, ...args: any[]) => void) | null | undefined} callback - callback function to be executed
+ * @param {Error | string | null | undefined} error - error which will be used by the callback function. If callback is not a function and
  * error is given, a rejected Promise is returned. If error is given but it is not an instance of Error, it is converted into one.
- * @param {...any} [args] - as many arguments as needed, which will be returned by the callback function or by the Promise
- * @returns {Promise} - if Promise is resolved with multiple arguments, an array is returned
+ * @param {any[]} args - as many arguments as needed, which will be returned by the callback function or by the Promise
+ * @returns {Promise<any>} - if Promise is resolved with multiple arguments, an array is returned
  */
 function maybeCallbackWithError(callback, error, ...args) {
-    if (error !== undefined && error !== null && !(error instanceof Error)) {
+    if (error != undefined && !(error instanceof Error)) {
         // if its not a real Error, we convert it into one
```

Comment:
`!= undefined` checks agains `null` and `undefined`.

Reply:
I know but, we had a realstes case about a month ago.. I would rather do it explicitly.

Reply:
ok fine with me